### PR TITLE
First user is Admin

### PR DESCRIPTION
**Pull request by skrix**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
+  rolify
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
-
```
Comment (ailecksandr) [inline app/models/user.rb]: empty line

File: app/models/user.rb [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
   validates :email, uniqueness: true
+  after_create :assign_role
```
Comment (ailecksandr) [inline app/models/user.rb]: empty line

File: app/models/role.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Role < ApplicationRecord
+  has_and_belongs_to_many :users, join_table: :users_roles
+
+  scopify
+  belongs_to :resource,
```
Comment (volodialeobit) [inline app/models/role.rb]: empty line

File: db/migrate/20180802111429_rolify_create_roles.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class RolifyCreateRoles < ActiveRecord::Migration[5.2]
+  def change
+    create_table(:roles) do |t|
+      t.string :name
+      t.references :resource, :polymorphic => true
```
Comment (volodialeobit) [inline db/migrate/20180802111429_rolify_create_roles.rb]: 1) align
2) `:polymorphic => true` => `polymorphic: true`
:face_with_head_bandage:

File: db/migrate/20180802111429_rolify_create_roles.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+      t.timestamps
+    end
+
+    create_table(:users_roles, :id => false) do |t|
```
Comment (volodialeobit) [inline db/migrate/20180802111429_rolify_create_roles.rb]: `id: false`

File: db/migrate/20180802111429_rolify_create_roles.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      t.references :user
+      t.references :role
+    end
+    
+    add_index(:roles, [ :name, :resource_type, :resource_id ])
```
Comment (volodialeobit) [inline db/migrate/20180802111429_rolify_create_roles.rb]: `%i[name resource_type resource_id]`

File: db/migrate/20180802111429_rolify_create_roles.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    add_index(:users_roles, [ :user_id, :role_id ])
```
Comment (volodialeobit) [inline db/migrate/20180802111429_rolify_create_roles.rb]: `%i[user_id role_id]`

File: db/migrate/20180802111429_rolify_create_roles.rb (hunk 6 above)
Comment (volodialeobit) [reply db/migrate/20180802111429_rolify_create_roles.rb]: why won't you just make it this way: `t.string :name, index: true` everywhere?

File: Gemfile [hunk 8]
```diff
@@ -47,7 +47,7 @@ group :development, :test do
   # RSpec
   gem 'rspec-rails', '~> 3.7'
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
-  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem 'byebug', platforms: %i[mri mingw x64_mingw]
```
Comment (ailecksandr) [inline Gemfile]: prefer `pry` over `byebug`
It causes less bugs in collaboration with server

File: app/models/user.rb [hunk 9]
```diff
@@ -1,10 +1,19 @@
...
          :recoverable, :rememberable, :trackable, :validatable
 
   validates :email, uniqueness: true
+
+  after_create :assign_role
+
+  private
+
+  def assign_role
+    add_role(:lunch_admin) if User.first.eql?(self)
```
Comment (ailecksandr) [inline app/models/user.rb]: `unless User.exists?` looks better)

**Pull request merged**